api/server/router/image/image_routes.go

```diff
@@ -489,6 +489,15 @@ func (ir *imageRouter) getImagesSearch(ctx context.Context, w http.ResponseWrite
 		return err
 	}
 
+	// "is-automated" is deprecated and was removed in 1.45.
+	// If the client is older than 1.45, remove the filter to avoid erroring out
+	// on older clients.
+	if versions.LessThan(httputils.VersionFromContext(ctx), "1.45") {
+		for _, v := range searchFilters.Get("is-automated") {
+			searchFilters.Del("is-automated", v)
+		}
+	}
+
```

Comment:
Following our discussion on Slack; I think the plan was to keep the filter, but it would not produce results; from the original PR;

> Consequently, searching for `is-automated=true` will yield no results.

And the change to have `IsAutomated` always being `false` would be unversioned;

> The deprecation is not versioned, and applies to all API versions.

Motivation there was that the deprecation correlates with the upstream (docker hub) service being in process of deprecating it; IIRC, the field is already out of date with reality, so results are not correct in many cases, thus always setting it to `false` makes sure that we don't return "bogus" results here.

Reply:
Yeah so with this PR the `IsAutomated` response will always be `false` (regardless of API version), but the `is-automated` filter will always produce an error on API >= 1.45 (it no longer makes sense, so we don't want newer clients to be using it while still keeping old clients working).